src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/GetConnectionStringParams.cs

```diff
@@ -15,11 +15,16 @@ public class GetConnectionStringParams
         /// </summary>
         public string OwnerUri { get; set; }
 
+        /// <summary>
+        /// Connection information of the connection
+        /// </summary>
+        public ConnectionDetails ConnectionDetails { get; set; }
```

Comment:
@alanrenmsft There isn't any common pattern we use for getting union types over to the C# land right? On the TS side we can just do `connectionContext: string | IConnectionInfo` where the user can pass in either the URI or the connection info object, but the only way I can think of here is to just have it be a generic object - but I don't think that'll work with the deserializing that needs to happen on this end. Am I missing anything you can think of?

Reply:
this is the place in Table Designer where we are not sure about the type of the property: https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/TableDesigner/Contracts/TableChangeInfo.cs#L33

but I guess it is different because the possible data types are all primitive types. for your case, I think you will need a JsonConverter: https://stackoverflow.com/questions/20432166/how-to-deserialize-a-json-property-that-can-be-two-different-data-types-using-js

Reply:
Doing that would require doing some refactoring of the base message parsing stuff, don't think that's worth it at this time. Having separate fields isn't the prettiest but this isn't really user facing so I think it's fine to just go for the separate fields for now.